# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/sandbox/win/src/policy_engine_params.h
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 2
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/sandbox/win/src/policy_engine_params.h
--- a/security/sandbox/chromium/sandbox/win/src/policy_engine_params.h	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/sandbox/win/src/policy_engine_params.h	Wed Jun 12 11:10:48 2019 +0100
@@ -59,7 +59,7 @@
 //  probably add other types as well.
 class ParameterSet {
  public:
-  ParameterSet() : real_type_(INVALID_TYPE), address_(NULL) {}
+  ParameterSet() : real_type_(INVALID_TYPE), address_(nullptr) {}
 
   // Retrieve the stored parameter. If the type does not match ulong fail.
   bool Get(uint32_t* destination) const {
@@ -89,16 +89,13 @@
   }
 
   // False if the parameter is not properly initialized.
-  bool IsValid() const {
-    return real_type_ != INVALID_TYPE;
-  }
+  bool IsValid() const { return real_type_ != INVALID_TYPE; }
 
  protected:
   // The constructor can only be called by derived types, which should
   // safely provide the real_type and the address of the argument.
   ParameterSet(ArgType real_type, const void* address)
-      : real_type_(real_type), address_(address) {
-  }
+      : real_type_(real_type), address_(address) {}
 
  private:
   // This template provides the same functionality as bits_cast but
@@ -124,56 +121,49 @@
   ParameterSetEx(const void* address);
 };
 
-template<>
+template <>
 class ParameterSetEx<void const*> : public ParameterSet {
  public:
-  ParameterSetEx(const void* address)
-      : ParameterSet(VOIDPTR_TYPE, address) {}
-};
-
-template<>
-class ParameterSetEx<void*> : public ParameterSet {
- public:
-  ParameterSetEx(const void* address)
-      : ParameterSet(VOIDPTR_TYPE, address) {}
+  ParameterSetEx(const void* address) : ParameterSet(VOIDPTR_TYPE, address) {}
 };
 
+template <>
+class ParameterSetEx<void*> : public ParameterSet {
+ public:
+  ParameterSetEx(const void* address) : ParameterSet(VOIDPTR_TYPE, address) {}
+};
 
-template<>
+template <>
 class ParameterSetEx<wchar_t*> : public ParameterSet {
  public:
-  ParameterSetEx(const void* address)
-      : ParameterSet(WCHAR_TYPE, address) {}
+  ParameterSetEx(const void* address) : ParameterSet(WCHAR_TYPE, address) {}
 };
 
-template<>
+template <>
 class ParameterSetEx<wchar_t const*> : public ParameterSet {
  public:
-  ParameterSetEx(const void* address)
-      : ParameterSet(WCHAR_TYPE, address) {}
+  ParameterSetEx(const void* address) : ParameterSet(WCHAR_TYPE, address) {}
 };
 
 template <>
 class ParameterSetEx<uint32_t> : public ParameterSet {
  public:
-  ParameterSetEx(const void* address)
-      : ParameterSet(UINT32_TYPE, address) {}
+  ParameterSetEx(const void* address) : ParameterSet(UINT32_TYPE, address) {}
 };
 
-template<>
+template <>
 class ParameterSetEx<UNICODE_STRING> : public ParameterSet {
  public:
-  ParameterSetEx(const void* address)
-      : ParameterSet(UNISTR_TYPE, address) {}
+  ParameterSetEx(const void* address) : ParameterSet(UNISTR_TYPE, address) {}
 };
 
 template <typename T>
 ParameterSet ParamPickerMake(T& parameter) {
   return ParameterSetEx<T>(&parameter);
-};
+}
 
 struct CountedParameterSetBase {
-  int count;
+  size_t count;
   ParameterSet parameters[1];
 };
 
@@ -185,15 +175,13 @@
 struct CountedParameterSet {
   CountedParameterSet() : count(T::PolParamLast) {}
 
-  ParameterSet& operator[](typename T::Args n) {
-    return parameters[n];
-  }
+  ParameterSet& operator[](typename T::Args n) { return parameters[n]; }
 
   CountedParameterSetBase* GetBase() {
     return reinterpret_cast<CountedParameterSetBase*>(this);
   }
 
-  int count;
+  size_t count;
   ParameterSet parameters[T::PolParamLast];
 };
 